### PR TITLE
Fixes #34556 - Developer guide

### PR DESCRIPTION
**Pull request by stejskalleos**

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+```
+leapp upgrade
+```
+
+## Resources and documentation
+
+* [Upgrading from RHEL 7 to RHEL 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/upgrading_from_rhel_7_to_rhel_8)
+* ~~[leapp-to.github.io/gettingstarted](https://leapp-to.github.io/gettingstarted)~~ is outdated, do not use!
+* [Developer documentation for Leapp tool](https://leapp.readthedocs.io/en/latest/index.html)
```
Comment (Ron-Lavi) [inline docs/developer_guide.md:75]: found another one specific to Satellite: https://access.redhat.com/documentation/en-us/red_hat_satellite/6.10/html/managing_hosts/upgrading-hosts-from-rhel7-to-rhel8_managing-hosts

File: docs/developer_guide.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Developer guide
+
+## Leapp tool
```
Comment (Ron-Lavi) [inline docs/developer_guide.md:3]: How about adding more info about what the plugin does, e.g:

The plugin enables the consumer to use a job template to upgrade Red Hat Enterprise Linux 7 hosts to Red Hat Enterprise Linux 8.

example usage:
- In the Foreman web UI, navigate to Hosts > All Hosts.
- Select the hosts that you want to upgrade to Red Hat Enterprise Linux 8.
- In the upper right of the Hosts window, from the Select Action list, select **Preupgrade check with Leapp**.
- Click Submit to start the pre-upgrade check.
- When the check is finished, click the **Leapp preupgrade report tab** to continue

Read more about it [here](https://access.redhat.com/documentation/en-us/red_hat_satellite/6.10/html/managing_hosts/upgrading-hosts-from-rhel7-to-rhel8_managing-hosts)

File: docs/developer_guide.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+yum-config-manager --add-repo https://copr.fedorainfracloud.org/coprs/g/oamg/leapp/repo/epel-7/group_oamg-leapp-epel-7.repo
+
+yum install yum-utils leapp
+```
+
+## Leapp commands
+
+**Pre-upgrade check**
+```
+leapp preupgrade
```
Comment (Ron-Lavi) [inline docs/developer_guide.md:54]: can you add the same warning here that currently there's an error with it and better try with RHEL?

File: docs/developer_guide.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+# => /var/log/leapp/leapp-report.json (This one is parsed by foreman-leapp)
+# => /var/log/leapp/leapp-report.txt
+# => /var/log/leapp/leapp-preupgrade.log
+
+leapp preupgrade --verbose  # Enable verbose logging
+leapp preupgrade --debug    # Enable debug mode
+```
+For the report sample see the file in `report_samples` folder.
```
Comment (Ron-Lavi) [inline docs/developer_guide.md:63]: should we copy the `report_samples/leapp-report.json` to `report_samples/leapp-report.json`

**Review by Ron-Lavi — COMMENTED** [review]
Thanks for writing this @stejskalleos 🙏🏼

File: docs/developer_guide.md (hunk 4 above)
Comment (stejskalleos) [reply docs/developer_guide.md:63]: Sorry I don't follow, both paths are same.

Comment (Ron-Lavi) [reply docs/developer_guide.md:63]: oops, copy pasting is hard, I meant:
should the developer copy the `report_samples/leapp-report.json` to
`/var/log/leapp/leapp-report.json` in order to use this mocked report?

File: docs/developer_guide.md (hunk 3 above)
Comment (stejskalleos) [reply docs/developer_guide.md:54]: I will remove the whole section with CentOS. We should test it only on RHEL systems anyway

**Comment by ares** [comment]
LGTM, anything else? I guess we could merge otherwise.

**Review by Ron-Lavi — APPROVED** [review]
Sure for not going back to it, LGTM
thanks @stejskalleos !

**Pull request merged**